woocommerce/woocommerce-ios
# CI: Migrate from Intel to Apple Silicon

## Description
This PR migrates WCiOS from the Intel CI stack over to our new Apple Silicon CI by upgrading to the Xcode 15.1 image, which is only available on Apple Silicon.

- It removes the `cache-builder.yml` pipeline because that isn't needed anymore with the changes in the Apple Silicon infrastructure tooling. 
- It removes the `git-s3-cache` plugin from the pipelines because that also isn't needed with the AS infrastructure
- Uses the correct whitespace character so that tests can pass (similar to https://github.com/woocommerce/woocommerce-ios/pull/11352/commits/c2a55f8585460fc991e7b97451af3b1608d01307)
-  Merging into `feat/xcode-15-ci` because @jaclync has been working on other Xcode 15-related fixes in that branch
- It's expected that the CI status checks for this PR will fail because those are still pointing to the Intel-based pipeline

Corresponding Terraform PR: https://github.com/Automattic/buildkite-ci/pull/339

## Testing instructions
It looks like there is a problem with the `test_load_support_screen` UI test: https://buildkite.com/automattic/woocommerce-ios-apple-silicon/builds/5

That looks similar to the other UI testing issues with Xcode 15 that @jaclync and @pachlava were discussing in https://github.com/woocommerce/woocommerce-ios/pull/11352, so I suggest we go ahead and merge this PR into that branch so they can continue working on a solution for all of the UI testing issues since those are an Xcode 15 problem and not an Apple Silicon CI problem. 

The "Build" and "Unit Tests" steps did complete successfully on the Apple Silicon CI 🎉 . You'll need to check the `WooCommerce iOS - Apple Silicon` pipeline manually because the regular `WooCommerce iOS` pipeline is pointing to the Intel nodes. https://buildkite.com/automattic/woocommerce-ios-apple-silicon/builds/

If more changes are made to this branch, you can go to that link and click "New Build". Under branch, type `move/wcios-to-apple-silicon-ci` and click "Create Build". That will kick off a build for this PR branch. 

## After Merging
- [ ] Update the description for https://github.com/woocommerce/woocommerce-ios/pull/11352 by adding these steps: 

```
When merging:
- [ ] Merge the corresponding Terraform PR to apply the configuration to the Buildkite pipeline: https://github.com/Automattic/buildkite-ci/pull/339

After merging:
- [ ] @spencertransier will post a P2 informing the WCiOS developers that they'll need to rebase on trunk in order for CI checks to pass on PRs 
```

---
